source/common/formatter/substitution_formatter.h

```diff
@@ -189,8 +191,13 @@ class StructFormatter {
   const bool omit_empty_values_;
   const bool preserve_types_;
   const std::string empty_value_;
+
+  // Note: don't use this ref outside of the constructor.
```

Comment:
Seems like we use it in `StructFormatter::toFormatStringValue`?

Reply:
correct, which is only called at construction time and never again -- so I want toa avoid a) transforming the commands std::vector from holding unique_ptrs to shared_ptrs and b) copying it which would be made possible by a).

Reply:
I guess the other option would be to pass it as a paramter to the helpers instead of storing it as member variable. It's a bit icky to have this field that might not be valid for the duration of the class

Reply:
We could pass it down the call chain.... Slightly more noisy but less icky than saving a ref that's only valid for a while, I agree.

Reply:
At the very least I would suggest emptying it out again after its no longer valid so that we're not keeping dangling references around only protected by a comment

Reply:
That I like -- it'd mean making the absl::optional non const... but that's probably OK.